File: example/claude/ts/quickstart/agent.ts
import {query} from "@anthropic-ai/claude-agent-sdk";
import * as readline from "readline";

// Agentic loop: streams messages as Claude works
// @ts-ignore
async function queryAgent(prompt: string) {
    // Wrap the for await loop in an async IIFE (Immediately Invoked Function Expression)
    // @ts-ignore
    await (async () => {
        // @ts-ignore
        for await (const message of query({
            prompt: prompt,
            options: {
                allowedTools: ["Read", "Edit", "Glob", "WebSearch", "Bash"],  // Tools Claude can use
                permissionMode: "acceptEdits",            // Auto-approve file edits
                systemPrompt: "你是一个AI助手，帮我用户完成任务",
            }
        })) {
            // Print human-readable output
            if (message.type === "assistant" && message.message?.content) {
                for (const block of message.message.content) {
                    if ("text" in block) {
                        console.log(block.text);             // Claude's reasoning
                    } else if ("name" in block) {
                        console.log(`Tool: ${block.name}`);  // Tool being called
                    }
                }
            } else if (message.type === "result") {
                console.log(`Done: ${message.subtype}`); // Final result
            }
        }
    })();
}

// Create readline interface to get user input
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("请输入您的提示 (prompt): ", (userPrompt) => {
    queryAgent(userPrompt);
    rl.close();
});

File: example/claude/py/quickstart/agent.py
import asyncio
import argparse
from claude_agent_sdk import query, ClaudeAgentOptions, AssistantMessage, ResultMessage


async def main(prompt: str):
    # Agentic loop: streams messages as Claude works
    async for message in query(
            prompt=prompt,
            options=ClaudeAgentOptions(
                allowed_tools=["Read", "Edit", "Glob", "WebSearch", "Bash"],  # Tools Claude can use
                permission_mode="acceptEdits",  # Auto-approve file edits
                system_prompt="你是一个AI助手，帮助用户完成任务",
            )
    ):
        # Print human-readable output
        if isinstance(message, AssistantMessage):
            for block in message.content:
                if hasattr(block, "text"):
                    print(block.text)  # Claude's reasoning
                elif hasattr(block, "name"):
                    print(f"Tool: {block.name}")  # Tool being called
        elif isinstance(message, ResultMessage):
            print(f"Done: {message.subtype}")  # Final result


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Claude-Agent CLI")
    parser.add_argument("prompt", nargs="?", default="", help="输入的prompt内容")
    args = parser.parse_args()

    print("=== 启动Claude-Agent ===")
    prompt = args.prompt if args.prompt else input("请输入prompt: ")
    asyncio.run(main(prompt))
